sdk/servicebus/service-bus/src/receivers/receiverCommon.ts

```diff
@@ -245,23 +246,25 @@ export function settleMessageOperation(
     throw error;
   }
 
-  // Message Settlement with managementLink
-  // 1. If the received message is deferred as such messages can only be settled using managementLink
-  // 2. If the associated receiver link is not available. This does not apply to messages from sessions as we need a lock on the session to do so.
-  if (isDeferredMessage || ((!receiver || !receiver.isOpen()) && !isDefined(message.sessionId))) {
-    return context
-      .getManagementClient(entityPath)
-      .updateDispositionStatus(message.lockToken!, operation, {
-        ...options,
-        associatedLinkName,
-        sessionId: message.sessionId
-      })
-      .catch((err) => {
-        throw translateServiceBusError(err);
-      });
-  }
+  try {
+    // Message Settlement with managementLink
+    // 1. If the received message is deferred as such messages can only be settled using managementLink
+    // 2. If the associated receiver link is not available. This does not apply to messages from sessions as we need a lock on the session to do so.
+    if (isDeferredMessage || ((!receiver || !receiver.isOpen()) && !isDefined(message.sessionId))) {
+      return context
+        .getManagementClient(entityPath)
+        .updateDispositionStatus(message.lockToken!, operation, {
+          ...options,
+          associatedLinkName,
+          sessionId: message.sessionId
+        });
+    }
 
-  return receiver!.settleMessage(message, operation, options).catch((err) => {
+    await receiver!.settleMessage(message, operation, options);
```

Comment:
We just got bit in event hubs by the non-null assertion. Could you check if receiver is undefined and throw if it is? That should slightly simplify your other if statements and let you get rid of the non-null assertions.

Reply:
Updated, let me know if that looks good.